src/bit-fields-macros/src/builder.rs

```diff
@@ -0,0 +1,697 @@
+    struct_member_fields_initialization: proc_macro2::TokenStream,
+    /// String used to form the display the bit field, the lines represent:
+    /// 1. Top border
+    /// 2. Bit numbers
+    /// 3. Border
+    /// 4. Field idents
+    /// 5. Border
+    /// 6. Field values
+    /// 7. Bottom border
+    /// Fmt values (since write doesn't work with in place ones)
+    display_string: MultiLineString,
+    /// String used to pass arguments for `std::fmt::Display` implementation.
+    display_fmt_string: proc_macro2::TokenStream,
+    /// Struct data type (e.g. `u8`)
+    data_type: DataType,
+    /// Struct data type token (e.g. `u8`)
+    struct_data_type: proc_macro2::Ident,
+    /// Struct identifier
+    struct_name: proc_macro2::Ident,
+}
+impl BitFieldBuilder {
+    /// Constructs new `BitFieldBuilder`.
+    pub fn new(
+        struct_name: proc_macro2::Ident,
+        struct_data_type: proc_macro2::Ident,
+        data_type: DataType,
+    ) -> Self {
+        Self {
+            #[cfg(feature = "flag_set")]
+            flag_matching_from_hashset: proc_macro2::TokenStream::new(),
+            #[cfg(feature = "flag_set")]
+            flag_setting_hashset: proc_macro2::TokenStream::new(),
+            #[cfg(feature = "field_map")]
+            field_matching_from_hashmap: proc_macro2::TokenStream::new(),
+            #[cfg(feature = "field_map")]
+            field_setting_hashmap: proc_macro2::TokenStream::new(),
+            #[cfg(feature = "set_theory")]
+            fields_superset_fn: quote! { true },
+            #[cfg(feature = "set_theory")]
+            fields_subset_fn: quote! { true },
+            #[cfg(feature = "set_theory")]
+            fields_disjoint_fn: quote! { false },
+            #[cfg(feature = "set_theory")]
+            fields_intersection_fn: proc_macro2::TokenStream::new(),
+            #[cfg(feature = "set_theory")]
+            fields_union_fn: proc_macro2::TokenStream::new(),
+            struct_doc_table_layout: String::from(
+                "///     <tr><th>Bit/s</th><th>Identifier</th><th>Description</th></tr>\n",
+            ),
+            struct_member_fields: proc_macro2::TokenStream::new(),
+            struct_member_fields_initialization: proc_macro2::TokenStream::new(),
+            #[rustfmt::skip]
+            display_string: MultiLineString::from("\
+                ┌───────\n\
+                │ \x1b[1mBit/s\x1b[0m \n\
+                ├───────\n\
+                │ \x1b[1mDesc\x1b[0m  \n\
+                ├───────\n\
+                │ \x1b[1mValue\x1b[0m \n\
+                └───────",
+            ),
+            display_fmt_string: proc_macro2::TokenStream::new(),
+            data_type,
+            struct_data_type,
+            struct_name,
+        }
+    }
+    /// Adds a bit field to the structure.
+    #[allow(clippy::integer_arithmetic)]
+    pub fn add_bit_field(&mut self, start: u8, rustdoc: &str, ident: &proc_macro2::Ident, end: u8) {
```

Comment:
nitpick: As a nuance of naming, functions that make use of streams generally follow a convention along the lines of `write*` or `read*`. I find this can be helpful to know what kind resources a given function/method is using.

Reply:
I don't have a strong opinion here but I find `add` to be more intuitive in that `add_bit_fields` adds (pushes) a bit flag element to the builder. In that sense it can be treated like adding a single value to a vector (despite underlying it writes to multiple token streams).